client/core/core.go

```diff
@@ -1385,6 +1385,21 @@ func (c *Core) loadWallet(dbWallet *db.Wallet) (*xcWallet, error) {
 	}
 	wallet.Wallet = w
 	wallet.connector = dex.NewConnectionMaster(w)
+
+	// Parse fee rate limit
+	// NOTE We parse fee rate limit 'feeratelimit' settings and store it on xcWallet
+	// unlike other settings which are being configured and set on each asset's
+	// ExchangeWallet instance (see above asset.Setup call) as it required on
+	// order placement time
+	feeLimit := walletCfg.Settings["feeratelimit"]
+	if feeLimit != "" {
+		uintLimit, err := strconv.ParseUint(feeLimit, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		wallet.feeRateLimit = uintLimit
+	}
```

Comment:
Roles are mixed here. These are wallet settings. `Core` doesn't deal with wallet settings. This needs to be handled in the **btc** and **dcr** packages.

Have the `ExchangeWallet` reject these in `FundOrder` based on the `asset.Order.DexConfig.MaxFeeRate`

Reply:
@chappjc preferred it to be on `xcWallet` instead of each asset's `ExchangeWallet` which was the first approach I considered.  The reason was that we need them on `Core`-level when placing the order.

I see your point regarding mixing roles here tho, as we have all other fields logic on ExchangeWallet level.

Reply:
Only Core deals with this setting though.  I don't see the point of giving this to the ExchangeWallet.

Reply:
Looks like moving them to `ExchangeWallet` would solve the blocking issue i'm currently having, we would be able
to store the defaults on `ExchangeWallet` without the user saving the settings, having them Settings only and parsing them on Core level would make forcing defaults a messy task.

Reply:
It would be nice to implement this in **core**, but we chose to use wallet settings instead since it's easier, so we should implement this in the wallet.

Reply:
Just as a general rule, `Core` should never try to interpret the `WalletConfig.Settings`.

Reply:
yep, using defaults already exists for other fields( e.g feeFallback: https://github.com/decred/dcrdex/blob/master/client/asset/dcr/dcr.go#L447-L451)

Reply:
If you want, but ExchangeWallet does not use this setting so it is confusing to have to define it in each backend's `Config` struct and then make accessors so Core can get it.  Why can't Core just pull it out of the map?

Reply:
If this is really a hang up for you guys, we shouldn't use ConfigOpts at all then.  This is completely unrelated to the ExchangeWallet.

Reply:
Hrmm, that's what it currently does, Settings doesn't seem to store the defaults, so in the above code:
`feeLimit := walletCfg.Settings["feeratelimit"] \ == ""`

Reply:
Sure but you can get the defaults from WalletInfo

Reply:
I agree that it would be confusing to store it on `ExchangeWallet` and use it only in Core tho,
Let me try to find out if I could somehow pull also the defaults in Core-level

Reply:
@chappjc good point, sounds reasonable to me!

Reply:
Please follow @buck54321's advice though.  I completely defer to him on decisions of wallet config.  I am just concerned that if we make ExchangeWallet handle the setting in it's own Config, then we have to make getter methods on the ExchangeWallets so that Core can get the value, even though ExchangeWallet doesn't use it internally.

Reply:
If we want to implement this in `Core`, we should, but I'm not looking to use a hybrid approach that breaks important conventions just for convenience. It's not really a wallet setting, but we've chose to make it a wallet setting for now, so let's use it like a wallet setting. It's also easier, I think. The wallet knows it has a value to use. `Core` would have to check every wallet's map for existence. And in principle, we don't control what the asset developer uses that settings key for. We would also be breaking conventions about who defines and interprets settings keys. Before, the wallet settings were for wallet specific settings that the asset developer needed. Now there are wallet settings whose definition and use are dictated by the **core** implementation.

If we choose to implement this in **core** later, we can switch it up. It's no harder to implement this as wallet settings, and it apparently simplifies the task. It appears that the hybrid solutions is a source of complication rather than a convenience anyway.

Reply:
> I am just concerned that if we make ExchangeWallet handle the setting in it's own Config, then we have to make getter methods on the ExchangeWallets so that Core can get the value, even though ExchangeWallet doesn't use it internally.

Core doesn't need the value, does it?

Reply:
It's not a wallet setting, but it is an asset-specific setting.  And because this *has to be implemented in Core*, I guess we need to do this the hard way, without having it in the ConfigOpts.